pedpy/methods/method_utils.py

```diff
@@ -330,9 +367,111 @@ def compute_neighbors(
+        polygons = frame_data[POLYGON_COL].to_numpy()
+
+        touching = shapely.dwithin(
+            polygons[:, np.newaxis], polygons[np.newaxis, :], 1e-9
+        )
+
+        # the peds are not neighbors of themselves
+        np.fill_diagonal(touching, False)
+
+        # Filter neighbor relationships based on the touching matrix
+        ids = frame_data[ID_COL].to_numpy()
+        row_idx, col_idx = np.where(
+            touching
+        )  # Get row and column indices of True values
+        id_column = ids[row_idx]  # Extract original IDs
+        neighbor_column = ids[col_idx]  # Extract corresponding neighbor IDs
+
+        # Create DataFrame for this frame's neighbors
+        frame_neighbors = pd.DataFrame(
+            {
+                ID_COL: id_column,
+                FRAME_COL: frame,
+                NEIGHBOR_ID_COL: neighbor_column,
+            }
+        )
+
+        # Append to the result list
+        neighbor_df.append(frame_neighbors)
+
+    if not neighbor_df:
+        return pd.DataFrame(columns=[ID_COL, FRAME_COL, NEIGHBOR_ID_COL])
+
+    # Concatenate all frames' data into a single DataFrame
+    return (
+        pd.concat(neighbor_df, ignore_index=True)
+        .sort_values(by=[FRAME_COL, ID_COL])
+        .reset_index(drop=True)
+    )
+
+
+def compute_neighbor_distance(
+    *,
+    traj_data: TrajectoryData,
+    neighborhood: pd.DataFrame,
+) -> pd.DataFrame:
+    """Compute the distance between the neighbors.
+
+    Computes the distance between the position of neighbors. As neighbors the
+    result of :func:`~compute_neighbors` with parameter :code:`as_list=False`.
+
+    Args:
+        traj_data (TrajectoryData): trajectory data
+        neighborhood (pd.DataFrame): DataFrame containing the columns 'id',
+            'frame' and 'neighbor_id'. The result of :func:`~compute_neighbors`
+            with parameter :code:`as_list=False` can be used here as input.
+
+    Raises:
+        ValueError: When passing a result of :func:`~compute_neighbors`
+            with parameter :code:`as_list=True`.
+
+    Returns:
+        DataFrame containing the columns 'id', 'frame', 'neighbor_id' and
+        'distance'.
+    """
+    if NEIGHBORS_COL in neighborhood.columns:
+        raise ValueError(
+            "For compute the distance between neighbors compute the "
+            "neighborhood with `as_list=False`."
+        )
+
+    neighbors_with_position = neighborhood.merge(
+        traj_data.data[[ID_COL, FRAME_COL, POINT_COL]],
+        on=[ID_COL, FRAME_COL],
+        how="left",
+    )
+
+    neighbors_with_position = neighbors_with_position.merge(
+        traj_data.data[[ID_COL, FRAME_COL, POINT_COL]],
+        left_on=[NEIGHBOR_ID_COL, FRAME_COL],
+        right_on=[ID_COL, FRAME_COL],
+        suffixes=("", "_neighbor"),
+    )
+
+    neighbors_with_position[DISTANCE_COL] = shapely.distance(
+        neighbors_with_position[POINT_COL],
+        neighbors_with_position["point_neighbor"],
```

Comment:
I don't like "point_neighbor". :-(
What about 
```
NEIGHBOR_POINT_COL = POINT_COL + "_neighbor"
```
and then use
```
neighbors_with_position[DISTANCE_COL] = shapely.distance(
    neighbors_with_position[POINT_COL],
    neighbors_with_position[NEIGHBOR_POINT_COL],
)
```

Reply:
This is generally possible, but in my opinion, not needed, as this column is only for internal usage inside the function. Even when `POINT_COL` might get renamed, it doesn't need to be reflected directly here. 
If you think this would increase readability for devs I will change it.